### PR TITLE
Create user model & controller

### LINKED ISSUE #4: [2.0pt] Generate User Model & Controller
- Create a User Model with the validations and associations with `Reservations`

- Create the user controller with the create and sign_in action

### PR DESCRIPTION
**Pull request by AndreaM2429**
Hello @NitBravoA92! 🙋🏽‍♀️ 
I'm sending this PR because I created a new feature that requires your review.

# **🎯 Milestones:** 

- Create a `User model` that contains:
    - Validations
        - `name` has to be present
        - `username` has to be present, has a maximum length of 25, and be unique
    - Associations
        - `has_many` association with `reservation model`

- Create a `User controller` to manage the following API actions:
    - `get`: show action in the controller that verifies if the user exists or not
    - `post`: create a new user and return a success status or a message with the specific error

## 🏁 General requirements 🏁:

❇️ Gitflow
❇️ No linter errors
❇️ Follow best practices

☆.｡.:･ﾟ☆.｡.:･ﾟ☆.｡.:･ﾟ☆.｡.:･ﾟ☆.｡.:･ﾟ☆.｡.:･ﾟ☆.｡.:･ﾟ☆☆.｡.:･ﾟ☆.｡.:･ﾟ☆.｡.:･ﾟ☆.｡.:･ﾟ☆.｡.:･ﾟ☆.｡.:･ﾟ☆.｡.:･ﾟ☆☆.｡.:*･ﾟ☆.

### ♻️ Changes applied:

- Delete the routes on the `routes.rb` file because it belongs to another task

🙇‍♀️🙇‍♀️Thanks in advance Carmen for taking the time and all your effort to review these new implementations. If any change is required or you can see any improvements in the code please let me know it.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -3,4 +3,9 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+
+  # API endpoints
+  namespace :api do
+    resources :users, param: :username, only: [:show, :create]
+  end
```
Comment (NitBravoA92) [inline config/routes.rb:10]: @AndreaM2429 I kindly suggest you remove the routes from this implementation as it belongs to another task.

**Review by NitBravoA92 — CHANGES_REQUESTED** [review]
## Hello @AndreaM2429    , :wave:

Good job so far :+1: 
There are some issues that you still need to work on to go to the next task but you are almost there!



## Highlights
- User Model Validations and Associations created correctly :heavy_check_mark:
- UsersController created correctly :heavy_check_mark:
- methods to create and validate users done correctly :heavy_check_mark:
- No linter error :heavy_check_mark: 

## Status: REQUIRED CHANGES ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @NitBravoA92  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

File: config/routes.rb (hunk 1 above)
Comment (AndreaM2429) [reply config/routes.rb]: @NitBravoA92 thanks for the observation, I forgot that it belongs to another task.

Comment (NitBravoA92) [reply config/routes.rb]: @AndreaM2429 No problem. Good job applying the suggested changes!

**Review by NitBravoA92 — APPROVED** [review]
# Hello @AndreaM2429! 👋

 ## Congratulations! 🎉  

Great Job I can see your effort 😄 👏 You've done it. Your feature is complete!  There is nothing else to say  👍🏻 ✅  🥳 🎉 

# STATUS: APPROVED ✅ 🍾 
### To Highlight 💯

- Good Job implementing the requested changes👍 
- No error linters 👍
 
## Congratulations! 🎉 

congrats

# Optional suggestions

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application.

Cheers, and Happy coding!👏👏

**Comment by AndreaM2429** [comment]
@NitBravoA92 thanks for your time and effort reviewed the changes 🙇🏽‍♀️

**Pull request merged**